client/state/data-layer/wpcom/videos/poster/index.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import { pick } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { VIDEO_EDITOR_UPDATE_POSTER } from 'state/action-types';
+import {
+	setPosterUrl,
+	showError,
+	showUploadProgress,
+} from 'state/ui/editor/video-editor/actions';
+
+/**
+ * Updates the poster for a video.
+ *
+ * @param  {Object} store  Redux store
+ * @param  {Object} action  Action object
+ * @param {Function} next  Dispatches to next middleware in chain
+ * @returns {Object} original action
+ */
+export const updatePoster = ( { dispatch }, action, next ) => {
+	if ( ! ( 'file' in action.params || 'atTime' in action.params ) ) {
+		return next( action );
+	}
+
+	const { atTime, file } = pick( action.params, [ 'atTime', 'file' ] );
+	const params = Object.assign(
+		{
+			apiVersion: '1.1',
+			method: 'POST',
+			path: `/videos/${ action.videoId }/poster`,
+		},
+		file && { formData: [ [ 'poster', file ] ] },
+		atTime !== undefined && { body: { at_time: atTime } },
+	);
+
+	dispatch( http( params, action ) );
+
+	return next( action );
+};
+
+export const updatePosterUrl = ( { dispatch }, action, next, { poster } ) => {
+	dispatch( setPosterUrl( poster ) );
```

Comment:
I guess `poster` is the name of the value coming from the API? it might help clarity here to rename in the arguments because it feels like `poster` ought to be an object here instead of a URL string…

```js
…, next, { poster as posterURL } ) => {
```